File: C++/Point.cpp
#include "Point.hpp"
#include <bits/stdc++.h>
using namespace std;
	
Point::Point(){             //ctor defult, absis = 0, ordinat = 0;
    absis = 0; 
    ordinat = 0;
}

//ctor user define
Point::Point(double newAbsis,double newOrdinat){
    absis = newAbsis;
    ordinat = newOrdinat;
}

//getter
double Point::getAbsis(){
    return absis;
}
double Point::getOrdinat(){
    return ordinat;
}

//setter
void Point::setAbsis(double newAbsis){
    absis = newAbsis;
}
void Point::setOrdinat(double newOrdinat){
    ordinat = newOrdinat;
}

void Point::geser(double deltaX, double deltaY){
    absis += deltaX;
    ordinat += deltaY;
}

double Point::hitungJarak(const Point& other){
    double dx = other.absis - absis;
    double dy = other.ordinat - ordinat;
    return sqrt(dx*dx + dy*dy);
}

void Point::PrintInfo(){
    cout<<"Absis = "<<absis<<" "<<"Ordinat = "<<ordinat<<endl;
}

File: Java/Point.java
import java.lang.Math;
public class Point implements Comparable{
    private int absis;
    private int ordinat;

    public Point(){
        absis = 0;
        ordinat = 0;
    }

    public Point(int _absis, int _ordinat){
        absis = _absis;
        ordinat = _ordinat;
    }

    public int getAbsis(){
        return absis;
    }

    public void setAbsis(int _absis){
        absis = _absis;
    }

    public int getOrdinat(){
        return ordinat;
    }

    public void setOrdinat(int _ordinat){
        ordinat = _ordinat;
    }

    public double hitungJarak(Point _point){
        int dx = this.absis - _point.absis;
        int dy = this.ordinat - _point.ordinat;
        return Math.sqrt(dx*dx + dy*dy);
    }

    public void PrintInfo(){
        System.out.println("(" + absis + "," + ordinat + ")");
    }
    @Override
    public int compareTo(Object P){
        Point P1 = (Point) P;
        if(absis == P1.absis && ordinat == P1.ordinat){
            return 0;
        }
        else if(absis < P1.absis && ordinat < P1.ordinat){
            return -1;
        }
        else{
            return 1;
        }
    }
    @Override
    public String toString(){
        return ("X = "+absis+" Y = "+ordinat);
    }
}